### ISSUE TITLE
Extension point org.eclipse.rap.http.registry.filters has no effect in 4.0 RC1

### ISSUE THREAD

**Issue by clompsy** [eclipse-rap/org.eclipse.rap#223]
Trying out RAP 4.0 RC1-20240828-1126, I ran into an issue migrating the Filter to the new Extension point.

So I have a simple jakarta.servlet.Filter implementation class called my.app.FilterServlet which just does a sysout:

```
package my.app;

import java.io.IOException;

import jakarta.servlet.Filter;
import jakarta.servlet.FilterChain;
import jakarta.servlet.FilterConfig;
import jakarta.servlet.ServletException;
import jakarta.servlet.ServletRequest;
import jakarta.servlet.ServletResponse;

public class FilterServlet implements Filter {

	public FilterServlet() {
	}

	@Override
	public void init(final FilterConfig arg0) throws ServletException {
	}

	@Override
	public void destroy() {
	}

	@Override
	public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain)
			throws IOException, ServletException {
		System.out.println("FilterServlet.doFilter()");
		chain.doFilter(request, response);
	}

}
```

Further I registered the filter as follows using the new extension carried over from equinox:
```
   	<extension point="org.eclipse.rap.http.registry.filters">
    	<filter 
       		alias="/dummy"
        	class="my.app.FilterServlet"
        	load-on-startup="true">
    	</filter>
	</extension>
```

The application works but the doFilter method of my FilterServlet is not called.
This used to work with RAP 3.24, just with the old name of the extension point (equinox instead of rap).
Am I missing something or did this functionality change somehow?

I checked the documentation and apart from it saying that the class has to implement javax.servlet.Filter (which is obviously incorrect for RAP 4.0/Jakarta EE), I don't see anything that I could do wrong here.

**Comment by ifurnadjiev** [comment]
Will check it ASAP.

**Comment by mknauer** [comment]
We have a "nightly" test build available ([rap-head-runtime-signed #3083](https://ci.eclipse.org/rap/view/RAP%20main%20branch/job/rap-head-runtime-signed/3083/)) that includes the fix for your problem. The build output (ZIP and p2) can be consumed by following the `Build Artifacts` link of the job. If you have time, we'd appreciate to get your feedback.

**Comment by clompsy** [comment]
I just switched to the build artifacts p2 URLs of that #3083 build and our filter worked as expected :)
So can confirm that the issue is resolved for us with your fixes. Thanks a lot for the quick response!
